### ISSUE TITLE
Option to choose image size in Cover and Image block

### ISSUE THREAD

**Issue by pedjas** [WordPress/gutenberg#52507]
## What problem does this address?

When Featured image is chosen for Post, full size image is linked.

When Cover or Image block is inserted in post or template and set to use Featured image, full size image is used. This happens even if actual view of image is much smaller.

For Image block there is option to additionally choose image size used so user must be aware that his site unnecessary loads full size images just shows them small and use this option to change that.

For cover block there is not option to choose image size. Full image size is used each time.

This is huge waste of resources on site. The first rule of using images is to reduce their pixel resolution to match actual size that should be displayed, not to load large image and reduce display size using CSS.

## What is your proposed solution?

User interface should warns users on this and help them easily notice discrepancy in physical and displayed image size and fix that.

For Cover images there should be option for user to choose size of image.

**Comment by hanneslsm** [comment]
I agree - I also have proposed that even for regular images the correct image size should be used automatically. Somehow same as for the cover block, as you described.
This is part of the Phase 3: https://make.wordpress.org/core/2023/07/07/media-library/
Since Phase 3 just has started I don't want you to have high hopes to see this implemented very fast. However, it's on the list.

**Comment by pedjas** [comment]
> I agree - I also have proposed that even for regular images the correct image size should be used automatically. Somehow same as for the cover block, as you described. This is part of the Phase 3: https://make.wordpress.org/core/2023/07/07/media-library/ Since Phase 3 just has started I don't want you to have high hopes to see this implemented very fast. However, it's on the list.

This is great!

**Comment by Luke-SF** [comment]
I don't see the option for image block to change from "large" to "full". What code enables that in the theme/Wordpress?

**Comment by annezazu** [comment]
Thanks for opening this. Good news! I'm closing this out both because it is a duplicate of https://github.com/WordPress/gutenberg/issues/19223 and https://github.com/WordPress/gutenberg/issues/58819 and because this option was added for the Cover block in this PR: https://github.com/WordPress/gutenberg/pull/62926
